docs/intro/install.rst

```diff
@@ -69,20 +69,25 @@ Overview
 Upload Elgg
 -----------
 
--  Download the `latest version of Elgg`_
--  Upload the ZIP file with an FTP client to your server
--  Unzip the files in your domain's document root (/home/username/www).
-
-.. _latest version of Elgg: http://elgg.org/download.php
-
-Elgg also supports installing from source with Composer:
+With Composer (recommended):
 
 .. code:: shell
 
     cd /path/to/wwwroot/
     composer self-update
     composer global require "fxp/composer-asset-plugin:~1.0"
-    composer create-project Elgg/Elgg:~2.0 .
+    composer create-project elgg/starter-project:dev-master .
+
+If you're not comfortable with Composer, you can install from a pre-packaged zip:
```

Comment:
To answer the question directly: it makes updates way easier. Want to upgrade all your plugins? One command. Want to upgrade just a single plugin? One command. This is important in order to get security updates out to folks or just to get everyone comfortable in the latest and greatest.

And if you want to use composer but don't want the latest and greatest, you use more restrictive version numbers to indicate that. Everyone's happy!


I could add a clarification that it is recommended for those comfortable with command line, as opposed to just "recommended" which might be implied to mean "for everyone."